modules/consentManagement.js

```diff
@@ -182,6 +192,138 @@ function lookupIabConsent(cmpSuccess, cmpError, hookConfig) {
+  // if the CMP is not found, the iframe function will call the cmpError exit callback to abort the rest of the CMP workflow
+  try {
+    ccpaFunction = window.__uspapi || utils.getWindowTop().__uspapi;
+  } catch (e) { }
+
+  if (utils.isFn(ccpaFunction)) {
+    ccpaFunction('getConsentData', null, callbackHandler.consentDataCallback);
+  } else if (inASafeFrame() && typeof window.$sf.ext.uspapi === 'function') {
+    callCcpaWhileInSafeFrame('getConsentData', callbackHandler.consentDataCallback);
+  } else {
+    // find the CMP frame
+    let f = window;
+    let ccpaFrame;
+    while (!ccpaFrame) {
+      try {
+        if (f.frames['__uspapiLocator']) ccpaFrame = f;
+      } catch (e) { }
+      if (f === window.top) break;
+      f = f.parent;
+    }
+
+    if (!ccpaFrame) {
+      return cmpError('CCPA not found.', hookConfig);
+    }
+
+    callCcpaWhileInIframe('getConsentData', ccpaFrame, callbackHandler.consentDataCallback);
+  }
+
+  function inASafeFrame() {
+    return !!(window.$sf && window.$sf.ext);
+  }
+
+  function callCcpaWhileInSafeFrame(commandName, callback) {
+    function sfCallback(msgName, data) {
+      if (msgName === 'cmpReturn') {
+        let responseObj = (commandName === 'getConsentData') ? data.vendorConsentData : data.vendorConsents;
+        callback(responseObj);
+      }
+    }
+
+    // find sizes from adUnits object
+    let adUnits = hookConfig.adUnits;
+    let width = 1;
+    let height = 1;
+    if (Array.isArray(adUnits) && adUnits.length > 0) {
+      let sizes = utils.getAdUnitSizes(adUnits[0]);
+      width = sizes[0][0];
+      height = sizes[0][1];
+    }
+
+    window.$sf.ext.register(width, height, sfCallback);
+    window.$sf.ext.ccpa(commandName);
+  }
+
+  function callCcpaWhileInIframe(commandName, ccpaFrame, moduleCallback) {
+    /* Setup up a __ccpa function to do the postMessage and stash the callback.
+      This function behaves (from the caller's perspective identicially to the in-frame __ccpa call */
+    window.__uspapi = function (cmd, ver, callback) {
+      let callId = Math.random() + '';
+      let msg = {
+        __uspapiCall: {
+          command: cmd,
+          version: ver,
+          callId: callId
+        }
+      };
+      uspapiCallbacks[callId] = callback;
+      ccpaFrame.postMessage(msg, '*');
+    };
+
+    /** when we get the return message, call the stashed callback */
+    window.addEventListener('message', readPostMessageResponse, false);
+
+    // call ccpa
+    window.__uspapi(commandName, 1, ccpaIframeCallback);
+
+    function readPostMessageResponse(event) {
+      let res = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
+      if (res.__uspapiReturn && res.__uspapiReturn.callId) {
+        let i = res.__uspapiReturn;
+        if (typeof uspapiCallbacks[i.callId] !== 'undefined') {
+          uspapiCallbacks[i.callId](i.returnValue, i.success);
```

Comment:
correctly passing both the `returnValue` and `success` params, but the callback only handles the first param.